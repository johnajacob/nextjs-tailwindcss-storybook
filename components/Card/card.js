export default function Card() {
  return (
    <div className="max-w-md mx-auto border-red-500 border-solid border">
      <h1 className="font-serif text-3xl">This is a title</h1>
      <p className="text-red-500">Abstract goes here</p>
    </div>
  )
}
