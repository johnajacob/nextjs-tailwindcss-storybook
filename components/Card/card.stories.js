import React from 'react'
import { action } from '@storybook/addon-actions'
import { Button } from '@storybook/react/demo'

export default { title: 'Button' }

export const withText = () => (
  <Button onClick={action('clicked')}>Hello Button</Button>
)

export const withSomeEmoji = () => (
  <Button onClick={action('clicked2')}>
    <span role="img" aria-label="so cool">
      😀 😎 👍 💯
    </span>
  </Button>
)
