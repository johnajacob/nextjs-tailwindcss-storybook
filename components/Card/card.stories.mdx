import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks';
import { withKnobs, select, boolean } from '@storybook/addon-knobs';

import Card from './card';

export const cardData =  {
  thumbImage: 'https://source.unsplash.com/Im7lZjxeLhg/640x480',
  primaryTopic: 'Gadgets',
  title: 'Using React Native for iOS and Android App Development at Walmart',
  abstract: 'We take a look back at what we saw in 2019, and think about what the next year might bring.',
  author: 'John Smith',
  publishedDate: 'Oct 14',
  commentsCount: 3
}

<Meta title="Design System|Card" component={Card} />

# Card

## Element to display content in a structured manner

<Preview withToolbar>
  <Story name="standard">
    <Card props={{...cardData}} />
  </Story>
</Preview>

<Props of={Card} />