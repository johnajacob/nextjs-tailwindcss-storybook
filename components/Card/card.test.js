/* eslint-env jest */

import React from 'react'
import { render } from '@testing-library/react'

import Card from './card'

describe('With React Testing Library', () => {
  it('Shows Card Component', () => {
    const { getByText } = render(<Card />)

    expect(getByText('This is a title')).not.toBeNull()
  })
})

describe('With React Testing Library Snapshot', () => {
  it('Should match Snapshot', () => {
    const { asFragment } = render(<Card />)

    expect(asFragment()).toMatchSnapshot()
  })
})
