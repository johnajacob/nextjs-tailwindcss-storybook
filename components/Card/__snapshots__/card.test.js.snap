// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library Snapshot Should match Snapshot 1`] = `
<DocumentFragment>
  <div
    class="max-w-md mx-auto border-red-500 border-solid border"
  >
    <h1
      class="font-serif text-3xl"
    >
      This is a title
    </h1>
    <p
      class="text-red-500"
    >
      Abstract goes here
    </p>
  </div>
</DocumentFragment>
`;
